packages/scripts/src/components/ExecuteButton.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <span v-if="loaded">
+    <b-button @click.stop="execute" :disabled="disabled"
+              :size="size" variant="primary"><font-awesome-icon class='mr-2' icon="play" size="lg" /> <slot></slot></b-button>
+    <RunModal v-model="showModal"
+              :parameters="parameters"
+              :name="name"
+    />
+  </span>
+</template>
+
+<script>
+import { mapState } from 'vuex'
+import RunModal from '../components/RunModal'
+
+export default {
+  name: 'ExecuteButton',
+  props: {
+    parameters: Array,
+    name: String,
+    form: Object,
+    size: {
+      type: String,
+      default: ''
+    },
+    disabled: {
+      type: Boolean,
+      default: false
+    },
+    doSave: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      showModal: false,
+      save: false
+    }
+  },
+  computed: {
+    ...mapState(['loaded'])
+  },
+  components: { RunModal },
+  methods: {
+    execute () {
+      if (this.doSave) {
+        this.$store.dispatch('saveParametersAndScripts', this.form).then(() => {
+          this.run()
+        })
+      } else {
+        this.run()
+      }
+    },
+    run () {
+      if (this.parameters.length > 0) {
+        this.showModal = true
+      } else {
+        window.open('/scripts/' + this.name + '/start', '_blank')
```

Comment:
as a user i really do not like this part , but i guess it is the same as it was before , so we should keep it "